cognite/client/utils/_pyodide_helpers.py

```diff
@@ -0,0 +1,230 @@
+
+
+def patch_sdk_for_pyodide() -> None:
+    # -------------------
+    # Patch Pyodide stuff
+    # - Patch 'requests' as it does not work in pyodide (socket not implemented):
+    import pyodide_http  # type: ignore [import]
+
+    pyodide_http.patch_all()
+
+    # - Patch the patched requests (from pyodide-http) as the SDK expects responses to have '.raw' attr:
+    pyo_http = pyodide_http._requests.PyodideHTTPAdapter
+    pyo_http._old_send = pyo_http.send
+    pyo_http.send = pyodide_http_adapter_send
+
+    # -----------------
+    # Patch Cognite SDK
+    # - For good measure ;)
+    os.environ["COGNITE_DISABLE_PYPI_VERSION_CHECK"] = "1"
+
+    # - Disable gzip, not supported:
+    cc.config.global_config.disable_gzip = True
+
+    # - Use another HTTP adapter:
+    cc._http_client.HTTPClient._old__init__ = cc._http_client.HTTPClient.__init__  # type: ignore [attr-defined]
+    cc._http_client.HTTPClient.__init__ = http_client__init__  # type: ignore [assignment]
+
+    # - If we are running inside of a JupyterLite Notebook spawned from Cognite Data Fusion, we set
+    #   the default config to FusionNotebookConfig(). This allows the user to:
+    #   >>> from cognite.client import CogniteClient
+    #   >>> client = CogniteClient()
+    if os.getenv("COGNITE_FUSION_NOTEBOOK") is not None:
+        cc.config.global_config.default_client_config = FusionNotebookConfig()
+
+    # - Inject these magic classes into the correct modules so that the user may import them normally:
+    cc.config.FusionNotebookConfig = FusionNotebookConfig  # type: ignore [attr-defined]
+    # TODO: Add once token refresh is working:
+    # cc.credentials.OAuthInteractiveFusionNotebook = OAuthInteractiveFusionNotebook  # type: ignore [attr-defined]
+
+    # - Set all usage of ThreadPoolExecutor to use a dummy, serial-implementation:
+    cc.utils._concurrency.ConcurrencySettings.executor_type = "mainthread"
+
+    # - Auto-ignore protobuf warning for the user (as they can't fix this):
+    warnings.filterwarnings(
+        action="ignore",
+        category=UserWarning,
+        message="Your installation of 'protobuf' is missing compiled C binaries",
+    )
+    # - AssetsAPI.create_hierarchy is not implemented yet (custom threading solution)
+    cc._api.assets._AssetPoster = NotImplementedAssetPoster  # type: ignore [assignment, misc]
+
+    # - DatapointsAPI.retrieve/retrieve_arrays/retrieve_dataframe:
+    cc._api.datapoints.PriorityThreadPoolExecutor = SerialPriorityThreadPoolExecutor  # type: ignore [assignment, misc]
+    cc._api.datapoints.as_completed = serial_as_completed  # type: ignore [assignment]
+
+
+def pyodide_http_adapter_send(self: Any, request: PreparedRequest, **kwargs: Any) -> Response:
+    # 'self' is actually `pyodide_http._requests.PyodideHTTPAdapter`, but we don't want it as a
+    # dependency just for the sake of mypy; after all, we are patching like monkeys 🙈
+    (response := self._old_send(request, **kwargs)).raw.version = ""
```

Comment:
I think this can be simplified by changing how we test for raw.version instead of patching this.

Reply:
I find one usage of it in `cognite/client/_api_client.py`:
`http_protocol_version = ".".join(list(str(res.raw.version)))`

It is used to print out a debug message, i.e. `HTTP/{http_protocol_version}` -> `HTTP/1.1`. Let's say I change it to a try-except, which protocol / what message should be logged instead?

Reply:
I'll push a change, then feel free to correct the wording correctness 😉

Reply:
I think that was the one that crashed (hence the fix), yep